File: app/src/main/java/me/zipi/navitotesla/AppExecutors.java
package me.zipi.navitotesla;

import java.util.concurrent.Executor;
import java.util.concurrent.SynchronousQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

public class AppExecutors {
    private static final Executor executor = new ThreadPoolExecutor(0, 50, 30, TimeUnit.SECONDS, new SynchronousQueue<>());


    private AppExecutors() {
    }

    public static void execute(Runnable command) {
        executor.execute(command);
    }

}

File: app/src/main/kotlin/me/zipi/navitotesla/AppExecutors.kt
package me.zipi.navitotesla

import java.util.concurrent.Executor
import java.util.concurrent.SynchronousQueue
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

object AppExecutors {
    private val executor: Executor =
        ThreadPoolExecutor(0, 50, 30, TimeUnit.SECONDS, SynchronousQueue())

    fun execute(command: Runnable) {
        executor.execute(command)
    }
}